src/printer/sygus_print_callback.cpp

```diff
@@ -78,8 +78,9 @@ void SygusExprPrintCallback::toStreamSygus(const Printer* p,
     sbody =
         sbody.substitute(vars.begin(), vars.end(), subs.begin(), subs.end());
 
+    // print to stream without letification
     std::stringstream body_out;
-    body_out << sbody;
+    p->toStream(body_out, sbody, -1, false, 0);
```

Comment:
I think this would be better handled if, by default, when doing SyGuS, the letification threshold would be 0 (by assigning the option to 0), but if the user were to set it to be bigger than we'd use that. Just because the default should be without lets doesn't mean a user can't want to print with them.

Reply:
We tried this actually.  The issue is you want the formulas printed by constraints to be letified (otherwise we can't convert large lustre benchmarks).  Thus, the letification threshold should determine this but always be overridden for grammar terms.

Reply:
I see. Please add this info to the PR description.